tests/fork-selection.rs

```diff
@@ -0,0 +1,598 @@
+//! Fork Selection Simulation
```

Comment:
What happened to the plan to move this to `book/src`, running assertions from markdown above, which would be run via `make -C book test`?

Reply:
lets do that in another PR, it wasn't clear what the Makefile should look like to run the cargo tests as well.